openmc/tallies.py

```diff
@@ -1963,8 +1964,12 @@ def _align_tally_data(self, other, filter_product, nuclide_product,
 
         Parameters
         ----------
+        other_old : openmc.Tally
+            The tally to outer product with this tally with the old tally
```

Comment:
I agree it can be confusing. How about:

            A copy of the tally to outer product with this tally. This version                                                                                                                                   
            retains the original alignment throughout the alignment procedure                                                                                                                                    
            in order and is needed to extract values with the old tally bin                                                                                                                                      
            ordering.